src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -259,16 +258,14 @@ private void generateStatusMessage(List<String> versionToQuery,
             }
             
             ActivityLogEntry activityLogEntry = new ActivityLogEntry(appLog);
-            activityLogEntry = data.filterLogs(activityLogEntry);
-            
-            boolean isToShow = activityLogEntry.toShow() && (!activityLogEntry.isTestingData() || data.getIfShowTestData());
-            if (!isToShow) {
+                       
+            if (!data.filterLogs(activityLogEntry)) {
                 continue;
             }
-            if (isFirstRow) {
-                activityLogEntry.setFirstRow();
-                isFirstRow = false;
+            if (activityLogEntry.isTestingData() && !data.getIfShowTestData()) {
```

Comment:
Both this block and the previous one only contains `continue`. See if you can combine the conditionals.